backend/src/v4/response_codes.js

```diff
@@ -490,6 +494,10 @@
 			// log bandwidth and http status code
 			systemLogger.logInfo(genResponseLogging(resCode, meta, req), undefined, logLabels.network);
 		}
+
+		if (killServer) {
```

Comment:
I'd make killServer a boolean eg

const killServer = resCode?.name === "MongoError" && resCode?.message?.includes && resCode.message.includes("Topology is closed");

and then do return Promise.reject(resCode);

just for clarity but its just preference

Reply:
@ChristosTsiotsias resCode gets reassigned right after I assigned `killServer` so the data would've been lost.

So to do what you want I'd have to assign 2 variables instead of one